framework/src/executioners/Transient.C

```diff
@@ -228,33 +228,36 @@ Transient::postStep()
 void
 Transient::execute()
 {
-
   preExecute();
 
+  if (_app.isRecovering())
+    incrementStepOrReject();
+
   // NOTE: if you remove this line, you will see a subset of tests failing. Those tests might have a wrong answer and might need to be regolded.
   // The reason is that we actually move the solution back in time before we actually start solving (which I think is wrong).  So this call here
   // is to maintain backward compatibility and so that MOOSE is giving the same answer.  However, we might remove this call and regold the test
   // in the future eventually.
-  if (!_app.isRecovering())
+  else
     _problem.advanceState();
 
+
   // Start time loop...
-  while (true)
+  while (keepGoing())
   {
-    if (_first != true)
-      incrementStepOrReject();
-
+#ifdef LIBMESH_ENABLE_AMR
+    if (!_first && _problem.adaptivity().isOn())
+      _problem.adaptMesh();
+#endif
```

Comment:
We may need a test if we don't have something similar in the suite. If you
want to gen one up great, if not I'll tag you when this gets closer to
getting merged so you'll have an opportunity to test it on your own code.

On Mon, May 9, 2016 at 12:15 PM Jesse Carter <notifications@github.com>
wrote:

> In framework/src/executioners/Transient.C
> <https://github.com/idaholab/moose/pull/6925#discussion_r62546099>:
>
> >    {
> > -    if (_first != true)
> > -      incrementStepOrReject();
> > -
> > +#ifdef LIBMESH_ENABLE_AMR
> > +    if (!_first && _problem.adaptivity().isOn())
> > +      _problem.adaptMesh();
> > +#endif
>
> Along the lines of what @andrsd <https://github.com/andrsd> said, I have
> a Marker that gets switched on and off at specific times and I wonder how
> this will affect it. For instance, if I detect that a nucleation event will
> occur on the next step, I refine the mesh to be ready for it. If adaptivity
> occurs *after* time is advanced, then I need to switch my marker to do
> the adaptivity if an event is occurring on *this* step.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6925/files/da663c1ca8be8b51f4977642357fc003f86b5a53#r62546099>
>